xmodule/services.py

```diff
@@ -215,12 +215,10 @@ def rebind_noauth_module_to_user(self, block, real_user):
             [
                 partial(DateLookupFieldData, course_id=self.course_id, user=self.user),
                 partial(OverrideFieldData.wrap, real_user, course),
-                partial(LmsFieldData, student_data=inner_student_data),
+                partial(LmsFieldData, student_data=student_data_real_user),
             ],
         )
 
-        block.scope_ids = block.scope_ids._replace(user_id=real_user.id)
```

Comment:
We don't need this - it's already handled in `bind_for_student`: https://github.com/openedx/edx-platform/blob/0b455e0336fa54199fd992c2ce83a81c07dd07c2/xmodule/x_module.py#L630-L631